src/usr/share/batwatch/batwatch.py

```diff
@@ -53,124 +46,150 @@ def __eq__(self, other):
+        self.logger.info('Monitoring the system\'s power state.')
 
         # Arbitrarily initialize the current status and detect any deviations.
-        current_status = CompositeStatus(1, FULLY_CHARGED)
+        prior_status = self._get_composite_status()
+
+        if prior_status.charge_status == DISCHARGING:
+            message = 'This system was discharging when Batwatch was initialized.'
+            self.logger.warning(message)
+            self._send_email(message)
 
         while True:
-            self.logger.trace('Running the main loop.')
-            new_status = self._get_composite_status()
-            if current_status != new_status:
-                self.logger.warn('A change in the battery status was detected. \nNEW STATUS: {} \nOLD STATUS: {}'
-                                 .format(current_status, new_status))
-                self._send_status_email(current_status, new_status)
-                current_status = new_status
+            current_status = self._get_composite_status()
+            if prior_status != current_status:
+                if prior_status < current_status:
+                    self.logger.warning('Battery state changed from %s to %s.',
+                                        prior_status, current_status)
+                else:
+                    self.logger.info('Battery state changed from %s to %s.', prior_status,
+                                     current_status)
+
+                self._send_email(self._build_email_body(prior_status, current_status))
+                prior_status = current_status
             else:
                 self.logger.trace('No changes in battery status.')
-            time.sleep(int(self.config['delay']))
 
-    def _get_composite_status(self):
-        """
-        Get status information about batteries connected to the device Batwatch is running on,
-         including the charge status, such that:
-        * If any battery is discharging, charge status is Discharging.
-          (Assume any status other than Charging or Fully Charged is equivalent to Discharging.)
-        * Charge status is Fully Charged only if all batteries are Fully Charged.
+            delay = random.uniform(0, self.config['average_delay'])
+            time.sleep(delay)
 
-        returns: CompositeStatus describing the count and overall charge status of the batteries.
+    def _get_composite_status(self):
+        """Get status information about batteries connected to the device Batwatch is running
+            on, including the charge status, such that:
+            * If any battery is discharging, charge status is Discharging.
+              (Assume any status other than Charging or Fully Charged is equivalent to
+               Discharging.)
+            * Charge status is Fully Charged only if all batteries are Fully Charged.
+
+        returns: CompositeStatus describing the count and overall charge status of the
+            batteries.
```

Comment:
Generally we don't reverse indent the paragraphs before the parameters. Also, we typically only reverse indent two spaces.

Reply:
The reverse indent should still only be two spaces.